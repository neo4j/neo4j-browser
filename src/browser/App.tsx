import { Auth0Provider } from '@auth0/auth0-react'
import { auth0Config } from 'shared/config/auth'

export function App() {
  return (
    <Auth0Provider {...auth0Config}>
      {/* existing app content */}
    </Auth0Provider>
  )
} 