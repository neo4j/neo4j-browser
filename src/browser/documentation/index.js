/*
 * Copyright (c) 2002-2019 "Neo4j,"
 * Neo4j Sweden AB [http://neo4j.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

// Help
import helpBolt from './dynamic/bolt'
import helpBoltEncryption from './help/bolt-encryption'
import helpBoltRouting from './help/bolt-routing'
import helpClear from './help/clear'
import helpContains from './help/contains'
import helpCreateConstraintOn from './help/create-constraint-on'
import helpCreateIndexOn from './help/create-index-on'
import helpCreate from './help/create'
import helpDelete from './help/delete'
import helpDropConstraintOn from './help/drop-constraint-on'
import helpDropIndexOn from './help/drop-index-on'
import helpDetachDelete from './help/detach-delete'
import helpEndsWith from './help/ends-with'
import helpExplain from './help/explain'
import helpForeach from './help/foreach'
import helpHistory from './help/history'
import helpHistoryClear from './help/history-clear'
import helpKeys from './help/keys'
import helpLoadCsv from './help/load-csv'
import helpMatch from './help/match'
import helpMerge from './help/merge'
import helpParam from './help/param'
import helpParams from './help/params'
import helpProfile from './help/profile'
import helpQueries from './help/queries'
import helpQueryPlan from './help/query-plan'
import helpRemove from './help/remove'
import helpRest from './help/rest'
import helpRestDelete from './help/rest-delete'
import helpRestGet from './help/rest-get'
import helpRestPost from './help/rest-post'
import helpRestPut from './help/rest-put'
import helpReturn from './help/return'
import helpSchema from './help/schema'
import helpServer from './help/server'
import helpServerUser from './help/server-user'
import helpSet from './help/set'
import helpStartsWith from './help/starts-with'
import helpStyle from './help/style'
import helpTemplate from './help/template'
import helpUnfound from './help/unfound'
import helpUnknown from './help/unknown'
import helpUnwind from './help/unwind'
import helpWhere from './help/where'
import helpWith from './help/with'

// Dynamic Help
import helpCommands from './dynamic/commands'
import helpCypher from './dynamic/cypher'
import helpHelp from './dynamic/help'
import helpPlay from './dynamic/play'

// Carousels
import guideConcepts from './guides/concepts'
import guideCypher from './guides/cypher'
import guideIntro from './guides/intro'
import guideLearn from './guides/learn'
import guideMovieGraph from './guides/movie-graph'
import guideNorthwindGraph from './guides/northwind-graph'

// Pages
import guideIconography from './guides/iconography'
import guideStart from './guides/start'
import guideTypography from './guides/typography'
import guideUnfound from './guides/unfound'
import guideWritecode from './guides/write-code'

export default {
  help: {
    title: 'Commands',
    chapters: {
      clear: helpClear,
      cypher: helpCypher,
      bolt: helpBolt,
      commands: helpCommands,
      guides: helpPlay,
      help: helpHelp,
      history: helpHistory,
      historyClear: helpHistoryClear,
      keys: helpKeys,
      play: helpPlay,
      server: helpServer,
      serverUser: helpServerUser,
      style: helpStyle,
      unfound: helpUnfound,
      unknown: helpUnknown,
      queries: helpQueries
    }
  },
  cypher: {
    title: 'Cypher',
    chapters: {
      contains: helpContains,
      createConstraintOn: helpCreateConstraintOn,
      createIndexOn: helpCreateIndexOn,
      create: helpCreate,
      delete: helpDelete,
      detachDelete: helpDetachDelete,
      dropConstraintOn: helpDropConstraintOn,
      dropIndexOn: helpDropIndexOn,
      endsWith: helpEndsWith,
      explain: helpExplain,
      foreach: helpForeach,
      loadCsv: helpLoadCsv,
      match: helpMatch,
      merge: helpMerge,
      param: helpParam,
      params: helpParams,
      profile: helpProfile,
      queryPlan: helpQueryPlan,
      remove: helpRemove,
      rest: helpRest,
      restDelete: helpRestDelete,
      restGet: helpRestGet,
      restPost: helpRestPost,
      restPut: helpRestPut,
      return: helpReturn,
      schema: helpSchema,
      set: helpSet,
      startsWith: helpStartsWith,
      template: helpTemplate,
      unwind: helpUnwind,
      where: helpWhere,
      with: helpWith
    }
  },
  bolt: {
    title: 'Bolt',
    chapters: {
      boltEncryption: helpBoltEncryption,
      boltRouting: helpBoltRouting
    }
  },
  play: {
    title: 'Guides & Examples',
    chapters: {
      concepts: guideConcepts,
      cypher: guideCypher,
      intro: guideIntro,
      learn: guideLearn,
      movieGraph: guideMovieGraph,
      movies: guideMovieGraph,
      northwindGraph: guideNorthwindGraph,
      northwind: guideNorthwindGraph,
      iconography: guideIconography,
      start: guideStart,
      typography: guideTypography,
      unfound: guideUnfound,
      writeCode: guideWritecode,
      // Commands only
      'query-template': {
        title: 'Query Templates',
        category: 'guides'
      }
    }
  }
}
