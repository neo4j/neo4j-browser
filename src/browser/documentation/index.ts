/*
 * Copyright (c) "Neo4j"
 * Neo4j Sweden AB [http://neo4j.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
// Help
import helpBolt from './dynamic/bolt'
// Dynamic Help
import helpCommands from './dynamic/commands'
import helpCypher from './dynamic/cypher'
import helpHelp from './dynamic/help'
import helpPlay from './dynamic/play'
import helpAlterUser from './help/alter-user'
import helpAuto from './help/auto'
import helpBoltEncryption from './help/bolt-encryption'
import helpBoltRouting from './help/bolt-routing'
import helpClear from './help/clear'
import helpContains from './help/contains'
import helpCreate from './help/create'
import helpCreateConstraintOn from './help/create-constraint-on'
import helpCreateDatabase from './help/create-database'
import helpCreateIndexOn from './help/create-index-on'
import helpCreateRole from './help/create-role'
import helpCreateUser from './help/create-user'
import helpDelete from './help/delete'
import helpDeny from './help/deny'
import helpDetachDelete from './help/detach-delete'
import helpDropConstraintOn from './help/drop-constraint-on'
import helpDropDatabase from './help/drop-database'
import helpDropIndexOn from './help/drop-index-on'
import helpDropRole from './help/drop-role'
import helpDropUser from './help/drop-user'
import helpEndsWith from './help/ends-with'
import helpExplain from './help/explain'
import helpForeach from './help/foreach'
import helpGrant from './help/grant'
import helpGrantRole from './help/grant-role'
import helpHistory from './help/history'
import helpHistoryClear from './help/history-clear'
import helpKeys from './help/keys'
import helpLoadCsv from './help/load-csv'
import helpMatch from './help/match'
import helpMerge from './help/merge'
import helpParam from './help/param'
import helpParams from './help/params'
import helpProfile from './help/profile'
import helpQueries from './help/queries'
import helpQueryPlan from './help/query-plan'
import helpRemove from './help/remove'
import helpRest from './help/rest'
import helpRestDelete from './help/rest-delete'
import helpRestGet from './help/rest-get'
import helpRestPost from './help/rest-post'
import helpRestPut from './help/rest-put'
import helpReturn from './help/return'
import helpRevoke from './help/revoke'
import helpRevokeRole from './help/revoke-role'
import helpSchema from './help/schema'
import helpServer from './help/server'
import helpServerUser from './help/server-user'
import helpSet from './help/set'
import helpShowDatabases from './help/show-databases'
import helpShowPrivileges from './help/show-privileges'
import helpShowRoles from './help/show-roles'
import helpShowUsers from './help/show-users'
import helpStartsWith from './help/starts-with'
import helpStyle from './help/style'
import helpTemplate from './help/template'
import helpUnfound from './help/unfound'
import helpUnknown from './help/unknown'
import helpUnwind from './help/unwind'
import helpWhere from './help/where'
import helpWith from './help/with'
// Play guides
import playConcepts from './play-guides/concepts'
import playCypher from './play-guides/cypher'
import playIconography from './play-guides/iconography'
import playIntro from './play-guides/intro'
import playLearn from './play-guides/learn'
import playMovieGraph from './play-guides/movie-graph'
import playNorthwindGraph from './play-guides/northwind-graph'
import playStart from './play-guides/start'
import playTypography from './play-guides/typography'
import playUnfound from './play-guides/unfound'
import playWritecode from './play-guides/write-code'
// Migrated sidebar guides
import guideConcepts from './sidebar-guides/concepts'
import guideCypher from './sidebar-guides/cypher'
import guideIntro from './sidebar-guides/intro'
import guideMovieGraph from './sidebar-guides/movie-graph'
import guideNorthwindGraph from './sidebar-guides/northwind-graph'
import guideUnfound from './sidebar-guides/unfound'

type AllDocumentation = {
  help: HelpDocs
  cypher: CypherDocs
  bolt: BoltDocs
  play: PlayDocs
  guide: GuideDocs
}
type DocItem = {
  title: string
  subtitle?: string
  category?: string
  content?: JSX.Element | null
  footer?: JSX.Element
  slides?: JSX.Element[]
}

export type Guide = {
  currentSlide: number
  title: string
  identifier: string
  slides: JSX.Element[]
  isError?: boolean
}

type GuideDocs = {
  title: 'Built-in Browser guides'
  chapters: Record<BuiltInGuideIdentifier, Omit<Guide, 'currentSlide'>>
}

export type BuiltInGuideIdentifier =
  | 'concepts'
  | 'cypher'
  | 'intro'
  | 'movie-graph'
  | 'movieGraph'
  | 'movies'
  | 'northwind'
  | 'northwind-graph'
  | 'northwindGraph'
  | 'unfound'

// TypeGuard function to ts to understand that a string is a valid key
export const isBuiltInGuide = (name: string): name is BuiltInGuideIdentifier =>
  name in docs.guide.chapters

type PlayDocs = {
  title: 'Guides & Examples'
  chapters: Record<PlayChapter, DocItem>
}

type PlayChapter =
  | 'concepts'
  | 'cypher'
  | 'iconography'
  | 'intro'
  | 'learn'
  | 'movie-graph'
  | 'movieGraph'
  | 'movies'
  | 'northwind'
  | 'northwind-graph'
  | 'northwindGraph'
  | 'start'
  | 'typography'
  | 'unfound'
  | 'writeCode'

export function isPlayChapter(name: string): name is PlayChapter {
  return name in docs.play.chapters
}

type BoltDocs = { title: 'Bolt'; chapters: Record<BoltChapter, DocItem> }
type BoltChapter = 'boltEncryption' | 'boltRouting'
type CypherDocs = { title: 'Cypher'; chapters: Record<CypherChapter, DocItem> }
type CypherChapter =
  | 'alterUser'
  | 'contains'
  | 'create'
  | 'createConstraintOn'
  | 'createDatabase'
  | 'createIndexOn'
  | 'createRole'
  | 'createUser'
  | 'delete'
  | 'deny'
  | 'detachDelete'
  | 'dropConstraintOn'
  | 'dropDatabase'
  | 'dropIndexOn'
  | 'dropRole'
  | 'dropUser'
  | 'endsWith'
  | 'explain'
  | 'foreach'
  | 'grant'
  | 'grantRole'
  | 'loadCsv'
  | 'match'
  | 'merge'
  | 'param'
  | 'params'
  | 'profile'
  | 'queryPlan'
  | 'remove'
  | 'rest'
  | 'restDelete'
  | 'restGet'
  | 'restPost'
  | 'restPut'
  | 'return'
  | 'revoke'
  | 'revokeRole'
  | 'schema'
  | 'set'
  | 'showDatabases'
  | 'showPrivileges'
  | 'showRoles'
  | 'showUsers'
  | 'startsWith'
  | 'template'
  | 'unwind'
  | 'where'
  | 'with'

type HelpDocs = { title: 'Commands'; chapters: Record<HelpChapter, DocItem> }
type HelpChapter =
  | 'auto'
  | 'bolt'
  | 'clear'
  | 'commands'
  | 'cypher'
  | 'guides'
  | 'help'
  | 'history'
  | 'historyClear'
  | 'keys'
  | 'play'
  | 'queries'
  | 'server'
  | 'serverUser'
  | 'style'
  | 'unfound'
  | 'unknown'

const docs: AllDocumentation = {
  help: {
    title: 'Commands',
    chapters: {
      auto: helpAuto,
      bolt: helpBolt,
      clear: helpClear,
      commands: helpCommands,
      cypher: helpCypher,
      guides: helpPlay,
      help: helpHelp,
      history: helpHistory,
      historyClear: helpHistoryClear,
      keys: helpKeys,
      play: helpPlay,
      queries: helpQueries,
      server: helpServer,
      serverUser: helpServerUser,
      style: helpStyle,
      unfound: helpUnfound,
      unknown: helpUnknown
    }
  },
  cypher: {
    title: 'Cypher',
    chapters: {
      alterUser: helpAlterUser,
      contains: helpContains,
      create: helpCreate,
      createConstraintOn: helpCreateConstraintOn,
      createDatabase: helpCreateDatabase,
      createIndexOn: helpCreateIndexOn,
      createRole: helpCreateRole,
      createUser: helpCreateUser,
      delete: helpDelete,
      deny: helpDeny,
      detachDelete: helpDetachDelete,
      dropConstraintOn: helpDropConstraintOn,
      dropDatabase: helpDropDatabase,
      dropIndexOn: helpDropIndexOn,
      dropRole: helpDropRole,
      dropUser: helpDropUser,
      endsWith: helpEndsWith,
      explain: helpExplain,
      foreach: helpForeach,
      grant: helpGrant,
      grantRole: helpGrantRole,
      loadCsv: helpLoadCsv,
      match: helpMatch,
      merge: helpMerge,
      param: helpParam,
      params: helpParams,
      profile: helpProfile,
      queryPlan: helpQueryPlan,
      remove: helpRemove,
      rest: helpRest,
      restDelete: helpRestDelete,
      restGet: helpRestGet,
      restPost: helpRestPost,
      restPut: helpRestPut,
      return: helpReturn,
      revoke: helpRevoke,
      revokeRole: helpRevokeRole,
      schema: helpSchema,
      set: helpSet,
      showDatabases: helpShowDatabases,
      showPrivileges: helpShowPrivileges,
      showRoles: helpShowRoles,
      showUsers: helpShowUsers,
      startsWith: helpStartsWith,
      template: helpTemplate,
      unwind: helpUnwind,
      where: helpWhere,
      with: helpWith
    }
  },
  bolt: {
    title: 'Bolt',
    chapters: {
      boltEncryption: helpBoltEncryption,
      boltRouting: helpBoltRouting
    }
  },
  play: {
    title: 'Guides & Examples',
    chapters: {
      concepts: playConcepts,
      cypher: playCypher,
      iconography: playIconography,
      intro: playIntro,
      learn: playLearn,
      movieGraph: playMovieGraph,
      'movie-graph': playMovieGraph,
      movies: playMovieGraph,
      northwind: playNorthwindGraph,
      'northwind-graph': playNorthwindGraph,
      northwindGraph: playNorthwindGraph,
      start: playStart,
      typography: playTypography,
      unfound: playUnfound,
      writeCode: playWritecode
    }
  },
  // Guides are play-guides but migrated to be viewable in the sidebar
  guide: {
    title: 'Built-in Browser guides',
    chapters: {
      concepts: guideConcepts,
      cypher: guideCypher,
      intro: guideIntro,
      movies: guideMovieGraph,
      movieGraph: guideMovieGraph,
      'movie-graph': guideMovieGraph,
      northwind: guideNorthwindGraph,
      northwindGraph: guideNorthwindGraph,
      'northwind-graph': guideNorthwindGraph,
      unfound: guideUnfound
    }
  }
}

export default docs
