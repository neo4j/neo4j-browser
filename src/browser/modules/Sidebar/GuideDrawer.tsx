/*
 * Copyright (c) "Neo4j"
 * Neo4j Sweden AB [http://neo4j.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
import React, { useRef } from 'react'
import { connect } from 'react-redux'
import { Action, Dispatch } from 'redux'

import { BackIcon } from 'browser-components/icons/LegacyIcons'

import GuideCarousel from '../GuideCarousel/GuideCarousel'
import GuidePicker from './GuidePicker'
import {
  BackIconContainer,
  StyledDrawerSeparator,
  StyledGuideDrawer,
  StyledGuideDrawerHeader
} from './styled'
import { Guide } from 'browser/documentation'
import { GlobalState } from 'shared/globalState'
import {
  RemoteGuide,
  fetchRemoteGuide,
  getCurrentGuide,
  getRemoteGuides,
  gotoSlide,
  resetGuide,
  setCurrentGuide,
  updateRemoteGuides
} from 'shared/modules/guides/guidesDuck'

export type GuideDrawerProps = {
  currentGuide: Guide | null
  remoteGuides: RemoteGuide[]
  backToAllGuides: () => void
  gotoSlide: (slideIndex: number) => void
  setCurrentGuide: (guide: Guide) => void
  fetchRemoteGuide: (identifier: string) => void
  updateRemoteGuides: (newList: RemoteGuide[]) => void
}

export const GuideDrawer = ({
  currentGuide,
  remoteGuides,
  backToAllGuides,
  gotoSlide,
  setCurrentGuide,
  fetchRemoteGuide,
  updateRemoteGuides
}: GuideDrawerProps): JSX.Element => {
  const scrollRef = useRef<HTMLDivElement>(null)

  return (
    <StyledGuideDrawer
      id="guide-drawer"
      data-testid="guidesDrawer"
      ref={scrollRef}
    >
      <StyledGuideDrawerHeader>
        {currentGuide !== null ? (
          <>
            <BackIconContainer
              data-testid="guidesBackButton"
              onClick={backToAllGuides}
            >
              <BackIcon width={16} />
            </BackIconContainer>
            {currentGuide.title}
          </>
        ) : (
          'Neo4j Browser Guides '
        )}
      </StyledGuideDrawerHeader>
      <StyledDrawerSeparator />
      {currentGuide === null ? (
        <GuidePicker
          remoteGuides={remoteGuides}
          setCurrentGuide={setCurrentGuide}
          fetchRemoteGuide={fetchRemoteGuide}
          updateRemoteGuides={updateRemoteGuides}
        />
      ) : (
        <GuideCarousel
          slides={currentGuide.slides}
          currentSlideIndex={currentGuide.currentSlide}
          gotoSlide={gotoSlide}
          scrollToTop={() =>
            scrollRef.current?.scrollIntoView({ block: 'start' })
          }
        />
      )}
    </StyledGuideDrawer>
  )
}

const mapStateToProps = (state: GlobalState) => ({
  currentGuide: getCurrentGuide(state),
  remoteGuides: getRemoteGuides(state)
})
const mapDispatchToProps = (dispatch: Dispatch<Action>) => ({
  backToAllGuides: () => dispatch(resetGuide()),
  gotoSlide: (slideIndex: number) => dispatch(gotoSlide(slideIndex)),
  setCurrentGuide: (guide: Guide) => dispatch(setCurrentGuide(guide)),
  fetchRemoteGuide: (identifier: string) =>
    dispatch(fetchRemoteGuide(identifier)),
  updateRemoteGuides: (updatedList: RemoteGuide[]) =>
    dispatch(updateRemoteGuides(updatedList))
})
const ConnectedGuideDrawer = connect(
  mapStateToProps,
  mapDispatchToProps
)(GuideDrawer)

export default ConnectedGuideDrawer
