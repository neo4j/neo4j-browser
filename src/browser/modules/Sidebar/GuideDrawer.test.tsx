/*
 * Copyright (c) "Neo4j"
 * Network Engine for Objects in Lund AB [http://neotechnology.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
import { render, screen } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore, { MockStoreEnhanced } from 'redux-mock-store'

import { GuideDrawer, GuideDrawerProps } from './GuideDrawer'
import { Guide } from 'browser/documentation'
import { RemoteGuide } from 'shared/modules/guides/guidesDuck'

const createProps = (
  currentGuide: Guide | null,
  remoteGuides: RemoteGuide[]
): GuideDrawerProps => ({
  currentGuide: currentGuide,
  remoteGuides: remoteGuides,
  backToAllGuides: () => undefined,
  gotoSlide: () => undefined,
  setCurrentGuide: () => undefined,
  fetchRemoteGuide: () => undefined,
  updateRemoteGuides: () => undefined
})

const mockStore = configureMockStore()

const renderWithRedux = (
  store: MockStoreEnhanced<unknown>,
  children: JSX.Element
) => {
  return render(<Provider store={store}>{children}</Provider>)
}

describe('GuideDrawer', () => {
  const store = mockStore({
    guides: {}
  })

  const remoteGuide: RemoteGuide = {
    currentSlide: 0,
    title: 'Title',
    identifier: 'identifier'
  }

  it('renders list view without Remote Guides', () => {
    const { container } = renderWithRedux(
      store,
      <GuideDrawer {...createProps(null, [])} />
    )

    expect(container).toMatchSnapshot()
  })

  it('renders list view including Remote Guides list', () => {
    const { container } = renderWithRedux(
      store,
      <GuideDrawer {...createProps(null, [remoteGuide])} />
    )

    expect(container).toMatchSnapshot()
    expect(screen.getByTestId('remoteGuidesTitle')).not.toBeNull()
    expect(screen.getByText('Title')).not.toBeNull()
  })

  it('renders guide slide when a guide is selected', () => {
    // scrollIntoView is not implemented in jsdom
    // Avoid TypeError: scrollIntoView is not a function
    window.HTMLElement.prototype.scrollIntoView = () => undefined

    const { container } = renderWithRedux(
      store,
      <GuideDrawer
        {...createProps(
          { ...remoteGuide, slides: [<div key={1}>Test Slide</div>] },
          [remoteGuide]
        )}
      />
    )

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Title')).not.toBeNull()
    expect(screen.getByText('Test Slide')).not.toBeNull()
  })
})
