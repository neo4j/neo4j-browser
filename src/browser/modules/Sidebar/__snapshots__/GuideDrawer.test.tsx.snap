// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuideDrawer renders guide slide when a guide is selected 1`] = `
<div>
  <div
    class="sc-efQSVx jRxDin"
    data-testid="guidesDrawer"
    id="guide-drawer"
  >
    <h4
      class="sc-cTAqQK dxVXPf"
    >
      <span
        class="sc-dPiLbb kcSurf"
        data-testid="guidesBackButton"
      >
        <i
          class="sc-bdvvtL kSEjuX"
        >
          <span
            class="SVGInline SVGInline--cleaned centeredSvgIcon"
          >
            <svg
              class="SVGInline-svg SVGInline--cleaned-svg centeredSvgIcon-svg"
              style="width: 16px;height: 16px;"
            />
          </span>
        </i>
      </span>
      Title
    </h4>
    <div
      class="sc-khQegj jfEAI"
    />
    <div
      class="sc-iUKqMP fBEGSa disable-font-ligatures"
    >
      <div>
        <div>
          Test Slide
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`GuideDrawer renders list view including Remote Guides list 1`] = `
<div>
  <div
    class="sc-efQSVx jRxDin"
    data-testid="guidesDrawer"
    id="guide-drawer"
  >
    <h4
      class="sc-cTAqQK dxVXPf"
    >
      Neo4j Browser Guides 
    </h4>
    <div
      class="sc-khQegj jfEAI"
    />
    <div
      class="sc-fWCJzd fWxPIy slide"
      style="padding: 0px 15px;"
    >
      You can also access Browser guides by running
      <span
        class="sc-iwjdpV iBEmuZ remove-play-icon"
        data-populate=":guide [guide name]"
      >
        :guide [guide name]
      </span>
      in the code editor.
      <div
        class="sc-cHzqoD kaPDph"
      >
        <div
          class="sc-dJjYzT gNzWMg"
        >
          Built-in guides
        </div>
      </div>
      <div
        class="sc-tAExr gRNbqr"
      >
        <li
          class="sc-hZpJaK cQWycM"
          data-testid="builtInGuideintro"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
          >
            :guide 
            intro
          </span>
          <div
            class="sc-cHzqoD kDRAfT"
          >
             
            Navigating Neo4j Browser
             
          </div>
        </li>
        <li
          class="sc-hZpJaK cQWycM"
          data-testid="builtInGuideconcepts"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
          >
            :guide 
            concepts
          </span>
          <div
            class="sc-cHzqoD kDRAfT"
          >
             
            Property graph model concepts
             
          </div>
        </li>
        <li
          class="sc-hZpJaK cQWycM"
          data-testid="builtInGuidecypher"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
          >
            :guide 
            cypher
          </span>
          <div
            class="sc-cHzqoD kDRAfT"
          >
             
            Cypher basics - create, match, delete
             
          </div>
        </li>
        <li
          class="sc-hZpJaK cQWycM"
          data-testid="builtInGuidemovie-graph"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
          >
            :guide 
            movie-graph
          </span>
          <div
            class="sc-cHzqoD kDRAfT"
          >
             
            Queries and recommendations with Cypher - movie use case}
             
          </div>
        </li>
        <li
          class="sc-hZpJaK cQWycM"
          data-testid="builtInGuidenorthwind-graph"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
          >
            :guide 
            northwind-graph
          </span>
          <div
            class="sc-cHzqoD kDRAfT"
          >
             
            Translate and import relation data into graph
             
          </div>
        </li>
      </div>
      <div
        class="sc-cHzqoD kaPDph"
      >
        <div
          class="sc-dJjYzT gNzWMg"
          data-testid="remoteGuidesTitle"
        >
          Remote Guides
        </div>
      </div>
      <div
        class="sc-tAExr gRNbqr"
      >
        <li
          class="sc-dkQkyq kFarBZ"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
            data-testid="remoteGuideidentifier"
          >
            Title
          </span>
          <span
            class="sc-gGPzkF btuKGg"
            data-testid="removeGuideidentifier"
          >
            <i
              class="sc-bdvvtL jIQruX sl-bin"
              style="line-height: inherit;"
            />
          </span>
        </li>
      </div>
      <div
        class="sc-JkixQ bmEYPU"
      >
        <a
          class="sc-bBHxTw hCOgTk"
          href="https://neo4j.com/graphgists/"
          rel="noreferrer"
          target="_blank"
        >
          More guides
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`GuideDrawer renders list view without Remote Guides 1`] = `
<div>
  <div
    class="sc-efQSVx jRxDin"
    data-testid="guidesDrawer"
    id="guide-drawer"
  >
    <h4
      class="sc-cTAqQK dxVXPf"
    >
      Neo4j Browser Guides 
    </h4>
    <div
      class="sc-khQegj jfEAI"
    />
    <div
      class="sc-fWCJzd fWxPIy slide"
      style="padding: 0px 15px;"
    >
      You can also access Browser guides by running
      <span
        class="sc-iwjdpV iBEmuZ remove-play-icon"
        data-populate=":guide [guide name]"
      >
        :guide [guide name]
      </span>
      in the code editor.
      <div
        class="sc-cHzqoD kaPDph"
      >
        <div
          class="sc-dJjYzT gNzWMg"
        >
          Built-in guides
        </div>
      </div>
      <div
        class="sc-tAExr gRNbqr"
      >
        <li
          class="sc-hZpJaK cQWycM"
          data-testid="builtInGuideintro"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
          >
            :guide 
            intro
          </span>
          <div
            class="sc-cHzqoD kDRAfT"
          >
             
            Navigating Neo4j Browser
             
          </div>
        </li>
        <li
          class="sc-hZpJaK cQWycM"
          data-testid="builtInGuideconcepts"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
          >
            :guide 
            concepts
          </span>
          <div
            class="sc-cHzqoD kDRAfT"
          >
             
            Property graph model concepts
             
          </div>
        </li>
        <li
          class="sc-hZpJaK cQWycM"
          data-testid="builtInGuidecypher"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
          >
            :guide 
            cypher
          </span>
          <div
            class="sc-cHzqoD kDRAfT"
          >
             
            Cypher basics - create, match, delete
             
          </div>
        </li>
        <li
          class="sc-hZpJaK cQWycM"
          data-testid="builtInGuidemovie-graph"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
          >
            :guide 
            movie-graph
          </span>
          <div
            class="sc-cHzqoD kDRAfT"
          >
             
            Queries and recommendations with Cypher - movie use case}
             
          </div>
        </li>
        <li
          class="sc-hZpJaK cQWycM"
          data-testid="builtInGuidenorthwind-graph"
        >
          <span
            class="sc-iwjdpV iBEmuZ remove-play-icon"
          >
            :guide 
            northwind-graph
          </span>
          <div
            class="sc-cHzqoD kDRAfT"
          >
             
            Translate and import relation data into graph
             
          </div>
        </li>
      </div>
      <div
        class="sc-JkixQ bmEYPU"
      >
        <a
          class="sc-bBHxTw hCOgTk"
          href="https://neo4j.com/graphgists/"
          rel="noreferrer"
          target="_blank"
        >
          More guides
        </a>
      </div>
    </div>
  </div>
</div>
`;
